starport/services/network/queries.go

```diff
@@ -0,0 +1,45 @@
+package network
+
+import (
+	"context"
+
+	launchtypes "github.com/tendermint/spn/x/launch/types"
+	"github.com/tendermint/starport/starport/pkg/events"
+	"github.com/tendermint/starport/starport/services/network/networkchain"
+)
+
+type LaunchInfo = networkchain.Launch
+
+// LaunchInfo fetches the chain launch from Starport Network by launch id.
+func (n Network) LaunchInfo(ctx context.Context, id uint64) (LaunchInfo, error) {
+	n.ev.Send(events.New(events.StatusOngoing, "Fetching chain information"))
+
+	res, err := launchtypes.NewQueryClient(n.cosmos.Context).Chain(ctx, &launchtypes.QueryGetChainRequest{
+		LaunchID: id,
+	})
+	if err != nil {
+		return LaunchInfo{}, err
+	}
+
+	n.ev.Send(events.New(events.StatusOngoing, "Chain information fetched"))
+
+	return networkchain.ParseLaunch(res.Chain), nil
+}
+
+// LaunchesInfo fetches the chain launches from Starport Network
+func (n Network) LaunchesInfo(ctx context.Context) ([]LaunchInfo, error) {
+	var launchesInfo []LaunchInfo
+
+	n.ev.Send(events.New(events.StatusOngoing, "Fetching chains information"))
+	res, err := launchtypes.NewQueryClient(n.cosmos.Context).ChainAll(ctx, &launchtypes.QueryAllChainRequest{})
+	if err != nil {
+		return launchesInfo, err
+	}
+
+	// Parse fetched chains
+	for _, chain := range res.Chain {
+		launchesInfo = append(launchesInfo, networkchain.ParseLaunch(chain))
+	}
+
```

Comment:
```suggestion
n.ev.Send(events.New(events.StatusDone, "Chains information fetched"))
```

Reply:
I think there are some refactoring we need to do first for the Event Emitter related to concurrency. If I include the `done` here it will be printed after the chain list
```
starport n --local chain list
Launch ID 	Chain ID 	Source 				Campaign ID
1 		earth-1 	https://github.com/lubtd/earth 	1
2 		earth-1 	https://github.com/lubtd/earth 	2
3 		toast-0 	test.com 			no campaign

✔ Chains information fetched
```

Reply:
You're right. In-fact we probably should not print StatusDone state on commands like list and show. And only print the StatusDone (permanent status texts) when we don't print a table with entrywriter to make terminal outputs clean.